wandb/apis/public.py

```diff
@@ -4314,8 +4316,8 @@ def from_id(cls, artifact_id: str, client: Client):
 
             artifact = cls(
                 client=client,
-                entity=None,
-                project=None,
+                entity=api.settings["entity"],
```

Comment:
This is not necessarily correct - the entity and project should be retrieved from the query, since it is not true the home entity/project is the entity/project fetching the artifact

Reply:
I think if you add this to the query, this PR can be reduced significantly

Reply:
Thanks @tssweeney. Changed in next commit.